projects.jsonl

```diff
@@ -0,0 +1,46 @@
s. The higher the number is, the more distinct the tokens are compared to the\nrest of the corpus.\n\nThese characteristics can give us a good intuition as to how well the SpanCat\nmodel identifies the correct spans. In the case of GENIA, the entities\nthemselves tend to be technical terms, which makes it more distinct and easier\nto classify. Again, we measure distinctiveness not only within the entities\nthemselves (SD), but also in its boundaries (BD).\n\nHere's some example data:\n\n![](./images/sample_00.png)\n![](./images/sample_01.png)\n\n\n### Experiments\n\nGiven what we know from the dataset, we will create the following pipelines:\n\n| Pipeline | Description                                                                                                                             | Workflow Name |\n|----------|-----------------------------------------------------------------------------------------------------------------------------------------|---------------|\n| SpanCat  | Pure Span Categorization for all types of entities. Serves as illustration to demonstrate suggester functions and as comparison to NER. | `spancat` |\n| NER      | Named-Entity Recognition for all types of entities. Serves as illustration to compare with the pure SpanCat implementation       | `ner`         |\n\n\n#### SpanCat Results\n\nBelow are the results for SpanCat. It seems that overall, span categorization\nis biased towards precision. This means that a large number of the suggested\nspans belong to the correct class. We can always tune how precise we want it\nto be: make the suggester lenient and we might get a lot of irrelevant hits,\nmake it strict and we miss might out on some true positives.\n\n|           | Precision   | Recall   | F-score   |\n|-----------|-------------|----------|-----------|\n| DNA       |    0.70     | 0.36     |  0.47     |\n| protein   |    0.77     | 0.52     |  0.62     |\n| cell_line |    0.77     | 0.30     |  0.44     |\n| cell_type |    0.76     | 0.62     |  0.68     |\n| RNA       |    0.77     | 0.25     |  0.38     |\n| **Overall**| 0.76       | 0.47     |  0.58     |\n\n#### NER Results\n\nNER performs well against SpanCat for all entity types, but note that this\nprocess entails training five (5) models per entity type. This might work if\nyou have a small number of entities, but can be computationally heavy if you\nhave a lot.\n\n|           |   Precision | Recall |   F-score |\n|-----------|-------------|--------|-----------|\n| DNA       |    0.74     | 0.63   |  0.68     |\n| protein   |    0.76     | 0.72   |  0.74     |\n| cell_line |    0.74     | 0.57   |  0.64     |\n| cell_type |    0.78     | 0.72   |  0.75     |\n| RNA       |    0.86     | 0.65   |  0.74     |\n\nSince we have five (5) separate models in NER, what we can do afterwards is\ncombine them into a single `Doc` that transfers `doc.ents` to `doc.spans`. Since\nthe tokens are the same, we don't need to worry about misalignments and the like.\n"}
```

Comment:
It seems to me that some of the "description" fields have been slightly "abused" to add all kinds of additional information that should only go in the readme. We should clean those out so that the resulting jsonl here is much more compact & clear.

Reply:
I think it's OK for the description to be long and README-like, though the inconsistency in the way we do this is definitely an issue. For now I'll modify a couple of READMEs to have less in the description and more in the non-automated part and see how that looks.

Reply:
For all of the project files where there was a lengthy description, I moved most of it to the actual README file. There are a few borderline ones, like `pipelines/polar_component`, where I just left them as-is, but they could be split up.

Doing this I can see why everything ends up in the project file - the auto-generated docs contain the title, which is a natural start, and end with a list of steps etc., which is nice to have but more supplementary. So in between those is a natural place to put docs, but it's not possible to do that with the auto generated docs unless everything goes in the project file. 

I think things are still OK with the changes here, but maybe something to think about for the future.